File: code-coaches/medium/how-far/HowFar.cs
using System;

class HowFar
{
    static int GetBlocksBetweenHouseAndPond(string layout)
    {
        // Find the indices of the money, thief, and guards on the casino floor
        int houseIndex = layout.IndexOf('H');
        int pondIndex = layout.IndexOf('P');
        int[] blockIndices = new int[100];
        int blockCount = 0;
        for (int i = 0; i < layout.Length; i++)
        {
            if (layout[i] == 'B')
            {
                blockIndices[blockCount++] = i;
            }
        }

        int count = 0;
        for (int i = 0; i < blockCount; i++)
        {
            if (Math.Min(houseIndex, pondIndex) < blockIndices[i] && blockIndices[i] < Math.Max(houseIndex, pondIndex))
            {
                count++;
            }
        }

        return count;
    }

    static void Main()
    {
        string layout = Console.ReadLine();
        Console.WriteLine(GetBlocksBetweenHouseAndPond(layout));
    }
}

File: code-coaches/medium/how-far/HowFar.java
import java.util.Scanner;

public class HowFar {

    public static int getBlocksBetweenHouseAndPond(String layout) {
        // Find the indices of the money, thief, and guards on the casino floor
        int houseIndex = layout.indexOf('H');
        int pondIndex = layout.indexOf('P');
        int[] blockIndices = new int[100];
        int blockCount = 0;
        for (int i = 0; i < layout.length(); i++) {
            if (layout.charAt(i) == 'B') {
                blockIndices[blockCount++] = i;
            }
        }

        int count = 0;
        for (int i = 0; i < blockCount; i++) {
            if (Math.min(houseIndex, pondIndex) < blockIndices[i] && blockIndices[i] < Math.max(houseIndex, pondIndex)) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String layout = scanner.nextLine();

        System.out.println(getBlocksBetweenHouseAndPond(layout));

        scanner.close();
    }
}
